WalletWasabi/CoinJoin/Client/Clients/CoinJoinClient.cs

```diff
@@ -584,8 +584,10 @@ private async Task TryRegisterCoinsAsync(ClientRound inputRegistrableRound)
 					// If our SatoshiClient does not yet know about the round, because of delay, then delay the round registration.
```

Comment:
I think L581 should be fixed too as `GetSingleOrDefaultRound` can return `null` too.

Reply:
Can you take a look at  https://github.com/zkSNACKs/WalletWasabi/pull/6726/commits/dea24065507ec31dbfca9991e7756c2d3adc32b4?

Reply:
I'm not sure about that condition https://github.com/zkSNACKs/WalletWasabi/commit/dea24065507ec31dbfca9991e7756c2d3adc32b4#diff-0312e8c17c589802ad123513e1028c8b2d27463c8adeb646ce0f80e214992f94R222 here. You know you have discovered an issue in the code so I think you should ask Lucas or David how to fix it imho. I'm not an author of the code and I don't know the design of that class well enough to help you here.

Reply:
According to @lontivero [comment](https://github.com/zkSNACKs/WalletWasabi/pull/6726/files#r753795660) it is correct.